pgcrypto_expressions/fields.py

```diff
@@ -71,18 +77,25 @@ def cached_col(self):
         return DecryptedCol(
             self.model._meta.db_table,
             self,
-            self.decrypt_sql,
+            self.decrypt_sql_template,
         )
 
 
 class DecryptedCol(Col):
-    def __init__(self, alias, target, decrypt_sql, output_field=None):
-        self.decrypt_sql = decrypt_sql
+    def __init__(self, alias, target, decrypt_sql_template, output_field=None):
+        self.decrypt_sql_template = decrypt_sql_template
+        self.target = target
         super(DecryptedCol, self).__init__(alias, target, output_field)
 
     def as_sql(self, compiler, connection):
+        key = self.target._get_encryption_key(connection)
+        decrypt_sql = self.decrypt_sql_template.format(
+            key=key,
+            dbtype=self.target.base_db_type,
+            sql='{sql}'
```

Comment:
This hack was only necessary because I was previously doing the first formatting step earlier, when the SQL was unavailable. Now you may as well move this format step after the `super` call and do everything at once, just including the SQL with the rest of the format data.